zulip/zulip/__init__.py

```diff
@@ -1270,7 +1270,8 @@ def add_subscriptions(self, streams: Iterable[Dict[str, Any]], **kwargs: Any) ->
             request=request,
         )
 
-    def remove_subscriptions(self, streams: Iterable[str], principals: Optional[Iterable[str]] = None) -> Dict[str, Any]:
+    def remove_subscriptions(self, streams: Iterable[str],
+                             principals: Optional[List[Union[str, int]]] = None) -> Dict[str, Any]:
```

Comment:
It should be a `Union[List`, not `List[Union`, since all the things in a list must be the same type.